interface/src/common/statusBar/statusBar.js

```diff
@@ -0,0 +1,12 @@
+angular.module('statusBar', [])
+
+.directive('statusBar', function() {
+  return {
+    restrict: 'E',
+    scope: {
+      msgText: '='
```

Comment:
I like this `<staus-bar>` gadget. Since **msgText** is read-only, we can use **@** instead of **=** here (then you have change the html above into:
```
<status-bar msgText="{{sample.status}}"></status-bar>
```

Another way is use the new **component()** function, which was introduced since Angular 1.5 for simple directives.
https://docs.angularjs.org/guide/component
(I am not saying that **component** is better than **directive** here, but I think it is good to know this option.)